### DISCUSSION TITLE
apple-iconが画面いっぱいに表示されない

### DISCUSSION THREAD

**Discussion by kagomen** [first-contributions-ja/first-contributions-ja.github.io#157] [10: 技術についての質問] [answered]
## 概要
apple-iconが画面いっぱいに表示されず、以下のようになってしまう
<img width="94" alt="スクリーンショット 2024-03-31 15 00 45" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/706e31fe-7123-4e6c-b8c4-9b82520e16d5">
大きめのサイズ変更しても、変化なしでした
みなさんのSafariでも同じでしょうか……？

## 関連イシュー
#156

**Comment by kazzyfrog** [comment]
ひょっとしたら、サイズは元のままで良くて、
自動で角が丸くなるので、正方形の画像じゃないといけなかったかも！？

https://note.com/nk2023/n/n345dca226533
> 画像に角丸等の処理はせず正方形で作る（ホーム画面に登録する際に自動で角丸等の処理が行われます）

**Comment by kazzyfrog** [comment]
ちなみに自分も同じような表示でした！

**Comment by kazzyfrog** [comment]
これたまに調べてますが、今のところわかってないです。
なにがダメなんだろ。。

**Comment by pss-aileen** [comment]
#148 これをやっても解決しない理由

### checkerで確認してみた
https://realfavicongenerator.net/favicon_checker?protocol=https&site=first-contributions-ja.github.io%2F

![スクリーンショット 2024-06-09 18 59 23](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/b4ff8eaf-e4df-4ac5-8a60-35fda15beab5)


### 気になるところ
- favicon.io が適応されている説？
- https://first-contributions-ja.github.io/icon-192x192.png こちらのURLにアクセスしても見れないから...？
- [The location of the apple-icon in the doc is not working properly. · Issue #52579 · vercel/next.js](https://github.com/vercel/next.js/issues/52579)

**Reply by kazzyfrog** [reply]
あ、そもそも`icon.svg`と`apple-icon.png`が、使われているっぽいですね！
192×192と、512×512は、どこでも使われていないやつですかね。。
https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons

サイトのheadはこんな感じ：
<img width="739" alt="スクリーンショット 2024-06-09 19 28 11" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/6c26db7d-368a-4734-a585-02764cb070d5">

**Reply by pss-aileen** [reply]
まってください、根本的に apple-icon.png 私変更できてませんでした...泣
なんかすっとぼけてました🙇‍♀️
すみません、今から変更します！

**Comment by pss-aileen** [comment]
### マスキング設定
`maskable`

- [Define your app icons - Progressive web apps | MDN](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/How_to/Define_app_icons#support_masking)
- [icons - ウェブアプリマニフェスト | MDN](https://developer.mozilla.org/ja/docs/Web/Manifest/icons)
- [Web Application Manifest](https://w3c.github.io/manifest/#dfn-maskable)
- [マスク可能なアイコンを使用する PWA のアダプティブ アイコンのサポート  |  Articles  |  web.dev](https://web.dev/articles/maskable-icon?hl=ja)

**Reply by pss-aileen** [reply]
![スクリーンショット 2024-06-09 21 29 24](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/ff582c63-dab3-4b05-a5f8-f3d61910bd7d)

**Reply by pss-aileen** [reply]
[PWAの App Icon 作成でデザイナーが知ったこと ++ Gaji-Laboブログ](https://www.gaji.jp/blog/2023/06/20/16156/)

この記事がわかりやすい...!?

**Reply by pss-aileen** [reply]
![スクリーンショット 2024-06-09 22 00 37](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/20c3da2e-a19e-4c63-817e-9a9822917211)

**Reply by pss-aileen** [reply]
自分のプロジェクトでmask-icon設定したらうまくいったので、manifest.json設定してみていいですか？
@kazzyfrog 

https://dev.to/progressier/why-a-pwa-app-icon-shouldnt-have-a-purpose-set-to-any-maskable-4c78

**Reply by kazzyfrog** [reply]
ぜひぜひ！
よろしくお願いします！！🙏

**Comment by pss-aileen** [comment, marked as answer]
こちら色々さらに調査したのですが、おそらく解決策わかりました。
apple-touch-iconで、「アルファチャンネルがはい」のものは、チェックしてみると周りに余白が表示されていました。

<img width="524" alt="スクリーンショット 2024-06-11 0 03 26" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/0fcc04e0-0370-45d9-982f-10522a68917c">

mac の safari で確認してたんですが、favicon test のものは アルファチャンネルなしのもので、余白が取り除かれました。

<img width="170" alt="スクリーンショット 2024-06-10 23 52 08" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/8120d0f5-017e-43bc-824f-0f51f86ba0b3">

昨日投稿していた、mask-iconについては safari のファビコンでsvgを設定するもので
manifest.jsonのほうは、android 系での設定なので、必要なさそうでした。
お騒がせしました🙇‍♀️

ということで、画像変更のissueとプルリク作成します！

**Reply by kazzyfrog** [reply]
アルファチャンネル知りませんでした！
勉強になりました！！

ただ、これでも、まだ変わらずって感じですかね。。！

**Reply by pss-aileen** [reply]
macのsafariは変わったんですが、iphoneのsafariはだめでした...🙇‍♀️
ちょっとお手上げです...

**Reply by kazzyfrog** [reply]
あ、でも確かにmacのsafariは変わってますね！！

ありがとうございます！！👍

**Reply by kazzyfrog** [reply]
iPhoneも、「ホームに追加」をすると正しい表示になっているので、
iphoneのsafariでも、いずれ反映されるかも！？です。。！

**Reply by pss-aileen** [reply]
iPadのsafariでも確認しましたが、領域いっぱいに広がっていました！
iPhoneでも反映されることを願います...！

**Reply by pss-aileen** [reply]
![image](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/f270aabd-8baf-483e-8f3e-1afc0ab9fb4e)
![image](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/d6802113-7c9e-4ae3-9fbc-d7e0b6db8448)

反映確認しました✨

**Reply by kazzyfrog** [reply]
おお！
原因は、アルファチャンネルだったんですね！

ありがとうございます！！🙌
